bindgen/ir/var.rs

```diff
@@ -30,6 +30,107 @@ pub(crate) enum VarType {
     String(Vec<u8>),
 }
 
+/// Numeric literal's radix.
+#[derive(Debug)]
+pub(crate) enum LiteralRadix {
+    /// Binary (base 2).
+    Binary,
+    /// Octal (base 8).
+    Octal,
+    /// Decimal (base 10).
+    Decimal,
+    /// Hexadecimal (base 16).
+    Hexadecimal,
+}
+
+/// Possible integer literal suffixes, all cases (U|u)?(L|l){0,2}, from longest to
+/// shortest in number of characters.
+#[rustfmt::skip] // hand-formatted for clarity
+const INTEGER_SUFFIXES: [&str; 21] = [
+    "ULL", "ULl", "UlL", "Ull", "uLL", "uLl", "ulL", "ull",
+    "UL", "Ul", "uL", "ul",
+    "LL", "Ll", "lL", "ll",
+    "U", "u",
+    "L", "l",
+    "",
+];
+
+impl LiteralRadix {
+    /// Obtain the number base of a bytestring corresponding to an existing integer
+    /// literal definition.
+    ///
+    /// Returns `None` if unable to infer a base.
+    pub(crate) fn from_integer_literal_token(
```

Comment:
Given that this function is supposed to work when _bytes corresponding to an existing integer literal_ have been passed to it, it's possibly doing far too much. I should look more into clang's API to see if all the extra parsing here is unnecessary. Same goes for the corresponding unit tests. If clang has already validated that the bytes represent an integer literal, determining the radix becomes much easier.